packages/eslint-plugin/src/rules/class-name-casing.ts

```diff
@@ -3,15 +3,19 @@
  * @author Jed Fox
  * @author Armano <https://github.com/armano2>
  */
-'use strict';
 
-const util = require('../util');
+import RuleModule from 'ts-eslint';
+import * as util from '../util';
+import { TSESTree } from '@typescript-eslint/typescript-estree';
 
 //------------------------------------------------------------------------------
 // Rule Definition
 //------------------------------------------------------------------------------
 
-module.exports = {
+type Options = [];
+type MessageIds = 'notPascalCased';
```

Comment:
is there any reason why we have to specify them by hand instead of using:
```ts
type MessageIds = Extract<keyof typeof rule.meta.messages, string>
```